### PR TITLE
[NEW] Chrome custom tabs support

### PR DESCRIPTION
**Pull request by bizzbyster**
@RocketChat/android

Closes https://github.com/RocketChat/Rocket.Chat.Android/issues/916

There are a number of customizations that we could consider but this PR should give us base support.

### CODE REVIEW

**Comment by bizzbyster** [comment]
Note: This is not ready to merge b/c building debug in circle.yml. Still figuring out how to do a release build in CircleCI on our fork.

**Comment by leonardoaramaki** [comment]
@bizzbyster Is this PR ready to be reviewed?

File: app/build.gradle [hunk 1]
```diff
@@ -110,6 +110,8 @@ dependencies {
     androidTestImplementation(libraries.expressoCore, {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
+
+    implementation 'com.android.support:customtabs:27.0.2'
```
Comment (leonardoaramaki) [inline app/build.gradle:114]: Next time please put the dependency on dependencies.gradle file! :)
I'll do this later.

**Review by leonardoaramaki — APPROVED** [review]

**Comment by bizzbyster** [comment]
@leonardoaramaki thanks for reviewing. @jaytat0 found an issue where Chrome Custom Tabs were not used when links found in the web preview were clicked. @Shailesh351 has since fixed that in another fork. I'll figure it out and add to PR today so that we merge this feature once and it covers all known cases.

**Comment by leonardoaramaki** [comment]
@bizzbyster I see. Already merged this one though so please open up another PR with next changes.

**Comment by bizzbyster** [comment]
Got it will do.

**Comment by bizzbyster** [comment]
@Shailesh351 any chance you can create a PR with your enhancement which also uses Chrome Custom Tabs for links within the web preview?

**Comment by bizzbyster** [comment]
@leonardoaramaki found a bug with this where if you click Http://www.cnn.com instead of http://www.cnn.com Rocket.chat crashes b/c the Chrome Custom Tab cannot handle the intent. I'll talk to @Shailesh and get a fix for this additional issue into his PR so we address both issues at once.

**Comment by Shailesh351** [comment]
@bizzbyster Change in https://github.com/RocketChat/Rocket.Chat.Android/blob/676b84c00ae482194fcdf3b2cca7aa3a389579dc/app/src/main/java/chat/rocket/android/chatroom/adapter/UrlPreviewViewHolder.kt#L36 with your custom tab implementation will work.

**Pull request merged**